pwem/tests/protocols/test_protocols_set_editor.py

```diff
@@ -0,0 +1,220 @@
+     [0., 0., 0., 1.]]
+]
+
+defocusList = [15000.,20000.,25000.]
+defocusAngle = [0., 10., 20.]
+projSize = 128
+samplingRate = 1.5
+
+class TestSets(pwtests.BaseTest):
+    """Run different tests related to the editor set protocol."""
+    @classmethod
+    def setUpClass(cls):
+        pwtests.setupTestProject(cls)
+
+    def _createSetOfParticles(self, setPartSqliteName, partFn,
+                             doCtf=False):
+        # create a set of particles
+
+        self.partSet = SetOfParticles(filename=setPartSqliteName)
+        self.partSet.setAlignment(ALIGN_PROJ)
+        self.partSet.setAcquisition(Acquisition(voltage=300,
+                                           sphericalAberration=2,
+                                           amplitudeContrast=0.1,
+                                           magnification=60000))
+        self.partSet.setSamplingRate(samplingRate)
+        if doCtf:
+            self.partSet.setHasCTF(True)
+        aList = [np.array(m) for m in mList]
+
+        for i, a in enumerate(aList):
+            p = Particle()
+            if doCtf:
+                defocusU = defocusList[i]
+                defocusV = defocusList[i]
+                ctf = CTFModel(defocusU=defocusU,
+                               defocusV=defocusV,
+                               defocusAngle=defocusAngle[i])
+                ctf.standardize()
+                p.setCTF(ctf)
+
+            p.setLocation(i + 1, partFn)
+            p.setTransform(emobj.Transform(a))
+            self.partSet.append(p)
+        self.partSet.write()
+
+    def importData(self, baseFn, objLabel, protType, importFrom):
+        prot = self.newProtocol(protType,
+                     objLabel=objLabel,
+                     filesPath=baseFn,
+                     maskPath=baseFn,
+                     sqliteFile=baseFn,
+                     haveDataBeenPhaseFlipped=False,
+                     magnification=10000,
+                     samplingRate=samplingRate,
+                     importFrom=importFrom
+                     )
+        self.launchProtocol(prot)
+        return prot
+
+    def testOperation(self):
+        """Make a trivial operation. Increment defocusU  by one"""
+        setPartSqliteName = self.proj.getTmpPath("particles_operate.sqlite")
+        setPartName = self.proj.getTmpPath('particles.stk')
+
+        self._createSetOfParticles(setPartSqliteName, setPartName,
+                                   doCtf=True)
+        protImportProj   = self.importData(setPartSqliteName,
+                                           "import projection\n operation",
+                                           ProtImportParticles,
+                                           ProtImportParticles.IMPORT_FROM_SCIPION)
+
+        #launch operate set protocol
+        protSetEditor = self.newProtocol(ProtSetEditor,
+                                         objLabel="operate")
+        protSetEditor.inputSet.set(protImportProj.outputParticles)
```

Comment:
Here, you can do 
protSetEditor.inputSet.set(protImportProj)
protSetEditor.inputSet.setExtended("outputParticles")